htdocs/product/class/product.class.php

```diff
@@ -3518,20 +3518,42 @@ function getLibFinished()
 	 * 	@param		string	$label			Label of stock movement
 	 * 	@param		double	$price			Unit price HT of product, used to calculate average weighted price (PMP in french). If 0, average weighted price is not changed.
 	 *  @param		string	$inventorycode	Inventory code
+	 *	@param		string	$origin_element	Origin element type
+	 *	@param		int		$origin_id		Origin id of element
 	 * 	@return     int     				<0 if KO, >0 if OK
 	 */
-	function correct_stock($user, $id_entrepot, $nbpiece, $movement, $label='', $price=0, $inventorycode='')
+	function correct_stock($user, $id_entrepot, $nbpiece, $movement, $label='', $price=0, $inventorycode='', $origin_element='', $origin_id=0)
 	{
 		if ($id_entrepot)
 		{
 			$this->db->begin();
 
+			if ($movement == 1)
```

Comment:
The PMP is only calculated on increase stock
When we do a negative movement the column price in the database isn't evaluate and I need to save the actual PMP to know which price it cost when is it use this price to show on project tab